sdks/java/io/mqtt/src/test/java/org/apache/beam/sdk/io/mqtt/MqttIOTest.java

```diff
@@ -193,6 +195,22 @@ public void run() {
     publisherThread.join();
   }
 
+  /**
+   * Test for BEAM-3282: this test should not timeout, and start/advance should return false.
+   */
+  @Test(timeout = 3 * 1000)
+  public void testReceiveWithTimeout() throws Exception {
+    pipeline.apply(MqttIO.read()
+        .withConnectionConfiguration(
+            MqttIO.ConnectionConfiguration.create(
+                "tcp://localhost:" + port,
+                "READ_TOPIC",
+                "READ_PIPELINE")).withMaxReadTime(Duration.standardSeconds(2)));
```

Comment:
This tests what happens if there's no data at all (start() returns false). How about also a test where there is some data?

Reply:
I added a test with `maxReadTime()` reading data.